File: backend/api/routes/stats.py
"""
Tubor Web — Routes API : statistiques & historique
"""

from fastapi import APIRouter, Query

from core.database import history
from models.schemas import (
    StatsSummaryResponse, DailyStatResponse,
    PlatformStatResponse, FormatStatResponse, HistoryItemResponse,
)

router = APIRouter(prefix="/api/stats", tags=["stats"])


@router.get("/summary", response_model=StatsSummaryResponse)
def stats_summary():
    return history.get_summary()


@router.get("/daily", response_model=list[DailyStatResponse])
def stats_daily(days: int = Query(14, ge=1, le=90)):
    return history.get_daily(days)


@router.get("/platforms", response_model=list[PlatformStatResponse])
def stats_platforms():
    return history.get_platforms()


@router.get("/formats", response_model=list[FormatStatResponse])
def stats_formats():
    return history.get_formats()


@router.get("/history", response_model=list[HistoryItemResponse])
def stats_history(
    limit:         int = Query(50,  ge=1,  le=200),
    offset:        int = Query(0,   ge=0),
    status_filter: str = Query(""),
    format_filter: str = Query(""),
):
    return history.get_history(
        limit=limit, offset=offset,
        status_filter=status_filter,
        format_filter=format_filter,
    )

File: frontend/src/api/stats.ts
import { api } from './client'
import type { StatsSummary, DailyStat, PlatformStat, FormatStat, HistoryItem } from '../types'

export const getStatsSummary = () => api.get<StatsSummary>('/stats/summary')
export const getDailyStats   = (days = 14) => api.get<DailyStat[]>(`/stats/daily?days=${days}`)
export const getPlatformStats = () => api.get<PlatformStat[]>('/stats/platforms')
export const getFormatStats  = () => api.get<FormatStat[]>('/stats/formats')
export const getHistory      = (params?: {
  limit?: number; offset?: number
  status_filter?: string; format_filter?: string
}) => {
  const q = new URLSearchParams()
  if (params?.limit)         q.set('limit',  String(params.limit))
  if (params?.offset)        q.set('offset', String(params.offset))
  if (params?.status_filter) q.set('status_filter', params.status_filter)
  if (params?.format_filter) q.set('format_filter', params.format_filter)
  return api.get<HistoryItem[]>(`/stats/history?${q}`)
}
